### DISCUSSION TITLE
Adding color profile fails

### DISCUSSION THREAD

**Discussion by trondhuso** [ImageMagick/ImageMagick#6142] [Help]
### ImageMagick version

6.9.10-23

### Operating system, version and so on

Linux Ubuntu Ubuntu 20.04.5 LTS

### Description
I am converting a PDF to a JPG and because of a requirement from a receiver of the JPG-file, I must add an icc color profile to the JPG-image.

I am running the following command to do this, and this has worked until last week (or that is what we think):
/usr/bin/convert /var/www/private/import/transform-folder/inputfile.pdf -flatten -set profile /usr/share/color/icc/ghostscript/default_gray.icc -colorspace RGB -resize 1023x1023 -density 400 -quality 100 /var/www/web/assets/pictures/outputfile.jpg

I am getting this message when running it and the file is not being produced:
convert: ../../magick/profile.c:1722: SetImageProfileInternal: Assertion `image->signature == MagickCoreSignature' failed

This bug just started to surface this last week (or when we got to know it)

Trond

**Comment by snibgo** [comment]
Options should be given in the same order that you want them to be executed. In particular, you shouldn't operate on an image before you have read it. Try this:
```
/usr/bin/convert inputFile -flatten -set profile /usr/share/color/icc/ghostscript/default_gray.icc -colorspace RGB -resize 1023x1023 -density 400 -quality 100 outputFile
```

**Reply by trondhuso** [reply]
Thanks for that correction @snibgo 
Same output though. but I like to do things correctly, so I have implemented this correction to the code.

As you can see (and I'll update the question) it transforms from a PDF-file to a JPG-file
/usr/bin/convert /var/www/private/import/transform-folder/inputfile.pdf -flatten -set profile /usr/share/color/icc/ghostscript/default_gray.icc -colorspace RGB -resize 1023x1023 -density 400 -quality 100 /var/www/web/assets/pictures/outputfile.jpg

**Comment by snibgo** [comment]
You get the same message `Assertion `image->signature == MagickCoreSignature' failed`? I don't know why that happens. It shouldn't. Please try:
```
/usr/bin/convert /var/www/private/import/transform-folder/inputfile.pdf  info:
```
This should write some text output. What is it?

Ideally, can you link to your input PDF? Then people can try to replicate the problem.

Does the problem occur only with that PDF, or also with other PDFs?

**Reply by trondhuso** [reply]
Output is: 
inputfile.pdf PDF 130x504 130x504+0+0 16-bit ColorSeparation CMYK 262181B 0.000u 0:00.009

I have to ask the client if it is OK to put the file here as it is a product they sell to other clients.

But I can probably ask them to create a test document and send that one over.

**Comment by snibgo** [comment]
>  ColorSeparation CMYK 

That could be the problem. You are trying to assign profile `default_gray.icc` to a CMYK image. This image has four channels, but I assume your profile is for one channel, or possibly three channels (R, G and B).

Try rasterizing the PDF as sRGB, by specifying that colorspace before reading it:
```
/usr/bin/convert -colorspace sRGB inputFile -flatten -set profile /usr/share/color/icc/ghostscript/default_gray.icc -colorspace RGB -resize 1023x1023 -density 400 -quality 100 outputFile
```
If that doesn't fix the problem, try also converting the image to grayscale before assigning the profile:
```
/usr/bin/convert -colorspace sRGB inputFile -flatten -colorspace Gray -set profile /usr/share/color/icc/ghostscript/default_gray.icc -colorspace RGB -resize 1023x1023 -density 400 -quality 100 outputFile
```

Later in your command, you convert to linear RGB with `-colorspace RGB`. There is nothing wrong with doing that, but perhaps you don't really want to convert to *linear* RGB.

(Very old versions of ImageMagick had `RGB` and `sRGB` the wrong way round, so `-colorspace RGB` would convert to non-linear sRGB.)
